rpc/errors.go

```diff
@@ -49,12 +49,15 @@ func Err(code int, data any) *RPCError {
 // Returns:
 // - error: the original error
 func tryUnwrapToRPCErr(err error, rpcErrors ...*RPCError) error {
-
 	var nodeErr *RPCError
 	if json.Unmarshal([]byte(err.Error()), nodeErr) != nil {
 		return err
 	}
 
+	if dataErr, ok := isErrorWithData(nodeErr); ok {
+		return dataErr
+	}
```

Comment:
The way this is handled now, if the error `err` is one of the listed errors with data in `isErrorWithData` method, it will be caught here even if it is not contained in `rpcErrors`. This is why I removed these errors from the call to `tryUnwrapToRPCErr` method throughout the code. I'm wondering if I should add them back for readability.
Also should I check the error with data in `isErrorWithData` method is contained in `rpcErrors` list? I didn't think it would cause any problems so I didn't add this verification, but wanted to double check

Reply:
> I'm wondering if I should add them back for readability.

I'm still learning the code base and could be wrong, but I think it is ok as it is in this PR.

> should I check the error with data in isErrorWithData method is contained in rpcErrors list?

I agree with what you did: I don't think the extra verification is necessary. To be honest, it might be simplest to define an rpcErrors slice and check them all every time, instead of explicitly passing a subset of errors as arguments. But that can be a separate PR.